bears/general/AnnotationBear.py

```diff
@@ -31,274 +37,50 @@ def run(self, filename, file, language: str, coalang_dir: str = None):
             ``u"string"``, the ``u`` will not be in the source range).
         """
         try:
-            lang_dict = LanguageDefinition(language, coalang_dir=coalang_dir)
-        except FileNotFoundError:
+            # ignore whitespace on pygments lexer name lookup
+            pygment = get_lexer_by_name(re.sub(r'\s+', '', language))
+        except ClassNotFound:
             content = ('coalang specification for ' + language +
                        ' not found.')
             yield HiddenResult(self, content)
             return
 
-        string_delimiters = dict(lang_dict['string_delimiters'])
-        multiline_string_delimiters = dict(
-            lang_dict['multiline_string_delimiters'])
-        multiline_comment_delimiters = dict(
-            lang_dict['multiline_comment_delimiters'])
-        comment_delimiter = dict(lang_dict['comment_delimiter'])
-        string_ranges = comment_ranges = ()
+        # the HiddenResult's content dict of SourceRange objects
+        content = {'comments': (), 'strings': ()}
+        # the pygments token types for the different types of source ranges
+        token_types = {'comments': Token.Comment, 'strings': Token.String}
```

Comment:
what we needed from the new annotationbear is that it should separately return ranges for singleline/multiline strings/comments.

Reply:
@aptrishu Can be next step... ;)